run.go

```diff
@@ -84,8 +86,26 @@ func TestingT(testingT *testing.T) {
 // provided run configuration.
 func RunAll(runConf *RunConf) *Result {
 	result := Result{}
-	for _, suite := range allSuites {
-		result.Add(Run(suite, runConf))
+	if !CustomParallelSuiteFlag {
+		for _, suite := range allSuites {
+			result.Add(Run(suite, runConf))
+		}
+		return &result
+	}
+
+	wg := sync.WaitGroup{}
+	rets := make([]*Result, len(allSuites))
+	for i, suite := range allSuites {
+		wg.Add(1)
+		rets[i] = &Result{}
+		go func(ret *Result, s interface{}) {
+			defer wg.Done()
+			ret = Run(s, runConf)
```

Comment:
assign the returned pointer to the input parameter is not correct, how about this:
```suggestion
			result.Add(Run(s, runConf))
```
and we can remove `rets`